### ISSUE TITLE
Consider only counting unique commits

### ISSUE THREAD

**Issue by mathbunnyru** [anuraghazra/github-readme-stats#2655]
### Describe the bug

I maintain a repository https://github.com/jupyter/docker-stacks/, which has 848 my commits.
Unfortunately, some users, instead of forking this repository, create their own copy.
This makes `github-readme-stats` count commits in this repos as well.

https://github-readme-stats-eight-theta.vercel.app/api?username=mathbunnyru&show_icons=true&theme=solarized-light&include_all_commits=true

This chart shows 68.6k commits, which would be awesome, but it's approximately 40x-80x larger than the correct number.
If I search my commits I see all these duplicates: https://github.com/search?o=desc&s=committer-date&type=Commits&q=author%3Amathbunnyru&p=94

### Expected behaviour

I expect `github-readme-stats` to parse the commits and only list the number of unique SHAs.
This way it would show the real number of commits.

### Screenshots / Live demo link

<img width="1470" alt="Screenshot 2023-04-27 at 15 33 08" src="https://user-images.githubusercontent.com/12270691/234849891-96a5db8b-62d1-468e-8147-558ae04a5807.png">
<img width="212" alt="Screenshot 2023-04-27 at 15 33 22" src="https://user-images.githubusercontent.com/12270691/234849919-47be7304-7a0b-4633-9e2a-95dc7627a485.png">

**Comment by rickstaa** [comment]
@mathbunnyru thanks for your bug report. I could reproduce the bug below. It is caused by the fact that we use the GitHub rest API for the total commits.

https://github.com/anuraghazra/github-readme-stats/blob/fd64333211fa9f39f59028fcb3892d2cf2fbb57a/src/fetchers/stats-fetcher.js#L146-L170

The behaviour is therefore as expected. You could create a pull request to improve the search query we use. We use the GitHub Rest API because of resource limitations in the GraphQL API (see https://docs.github.com/en/graphql/overview/resource-limitations). Fetching all commits through the GraphQL API, although more accurate, will result is in hitting the rate limiter faster. https://github.com/anuraghazra/github-readme-stats/pull/1691 will fix this, but we cannot merge it into the main branch because of this. You can, however, already use that pull request on your own Vercel instance, since there you will likely not hit the GraphQL limits. We will likely release it under our new GitHub action (see https://github.com/anuraghazra/github-readme-stats/issues/2179). 

### Reproduce

[![Mathbunnyru GitHub stats](https://github-readme-stats.vercel.app/api?username=Mathbunnyru)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Mathbunnyru GitHub stats](https://github-readme-stats.vercel.app/api?username=Mathbunnyru)](https://github.com/anuraghazra/github-readme-stats)
``` 

[![Mathbunnyru GitHub stats](https://github-readme-stats.vercel.app/api?username=Mathbunnyru&include_all_commits=true)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Mathbunnyru GitHub stats](https://github-readme-stats.vercel.app/api?username=Mathbunnyru&include_all_commits=true)](https://github.com/anuraghazra/github-readme-stats)
```

**Comment by mathbunnyru** [comment]
Thank you! I think you meant `https://github-readme-stats-rongronggg9.vercel.app` for the first chart, and I confirm that it works fine and the number of commits is as expected.

**Comment by rickstaa** [comment]
> Thank you! I think `https://github-readme-stats-rongronggg9.vercel.app` works well, and the number of commits is as expected.

@mathbunnyru please don't use @Rongronggg9 Vercel instance as you will likely cause the GraphQL resource limits to be reached 😅. I would like to recommend you to create your own Vercel instance and merge https://github.com/anuraghazra/github-readme-stats/pull/1691 into your main branch (see https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own).

**Comment by mathbunnyru** [comment]
Ok, sorry @rickstaa, I will do what you suggest 😅

**Comment by rickstaa** [comment]
> Ok, sorry @rickstaa, I will do what you suggest 😅

No problem, the GraphQL limit is a real bottleneck and not well explained in the README. That's why we want to switch to a GitHub action approach (see https://github.com/anuraghazra/github-readme-stats/issues/2179).

**Comment by mathbunnyru** [comment]
@rickstaa I don't think this is fixed yet.

**Comment by rickstaa** [comment]
> @rickstaa I don't think this is fixed yet.

You're right, I have mistaken it with another issue related to the language results 👍🏻.

**Comment by rickstaa** [comment]
You can also show your support for https://github.com/orgs/community/discussions/35675 which would allow us to solve this for the Vercel version.
